src/FirstSteps/RPG/Heroes/Hero.cs

```diff
@@ -100,11 +104,29 @@ public HeroModel ToHeroModel()
                     {
                         Name = item.Name,
                         Price = item.Price,
-                        Weight = item.Weight
+                        Weight = item.Weight,
                     }).ToList()
-                }
+                },
             };
             return heroModel;
+
+        }
+        public bool Resurrect()
+        {
+            if (Coins >= 5)
+            {
+                int maxHp =  _maxHealth / 2;
+                _health = maxHp;
+                Coins -= 5;
```

Comment:
Inside `Hero` you can access private fields like `_coinsBag`. (When you switch you will be able to remove the setter in Conins as well)